diff --git a/win32/detectenv-msvc.mak b/win32/detectenv-msvc.mak
index 80a5eed..926a9b2 100644
--- a/win32/detectenv-msvc.mak
+++ b/win32/detectenv-msvc.mak
@@ -132,7 +132,7 @@ CFLAGS = $(CFLAGS_ADD) /W3 /Zi /I.. /I..\src /I. /I$(PREFIX)\include
 !if "$(ADDITIONAL_LIB_DIR)" != ""
 ADDITIONAL_LIB_ARG = /libpath:$(ADDITIONAL_LIB_DIR)
 !endif
-LDFLAGS_BASE = $(LDFLAGS_ARCH) /libpath:$(PREFIX)\lib $(ADDITIONAL_LIB_ARG) /DEBUG
+LDFLAGS_BASE = $(LDFLAGS_ARCH) /libpath:$(PREFIX)\lib $(ADDITIONAL_LIB_ARG) /DEBUG /APPCONTAINER "WindowsApp.lib"
 
 !if "$(CFG)" == "debug"
 LDFLAGS = $(LDFLAGS_BASE)
