diff --git a/crypto/rand/rand_win.c b/crypto/rand/rand_win.c
index cf563ef..d01fd99 100644
--- a/crypto/rand/rand_win.c
+++ b/crypto/rand/rand_win.c
@@ -445,7 +445,7 @@ int RAND_poll(void)
                 ZeroMemory(&hlist, sizeof(HEAPLIST32));
                 hlist.dwSize = sizeof(HEAPLIST32);
                 if (good)
-                    starttime = GetTickCount();
+                    starttime = (DWORD)GetTickCount64();
 #   ifdef _MSC_VER
                 if (heaplist_first(handle, &hlist)) {
                     /*
@@ -474,7 +474,7 @@ int RAND_poll(void)
                                     RAND_add(&hentry, hentry.dwSize, 5);
                                 while (heap_next(&hentry)
                                        && (!good
-                                           || (GetTickCount() - starttime) <
+                                           || ((DWORD)GetTickCount64() - starttime) <
                                            MAXDELAY)
                                        && --entrycnt > 0);
                             }
@@ -488,7 +488,7 @@ int RAND_poll(void)
                         }
                     } while (heaplist_next(handle, &hlist)
                              && (!good
-                                 || (GetTickCount() - starttime) < MAXDELAY)
+                                 || ((DWORD)GetTickCount64() - starttime) < MAXDELAY)
                              && ex_cnt_limit > 0);
                 }
 #   else
@@ -507,7 +507,7 @@ int RAND_poll(void)
                         }
                     } while (heaplist_next(handle, &hlist)
                              && (!good
-                                 || (GetTickCount() - starttime) < MAXDELAY));
+                                 || ((DWORD)GetTickCount64() - starttime) < MAXDELAY));
                 }
 #   endif
 
@@ -520,13 +520,13 @@ int RAND_poll(void)
                 p.dwSize = sizeof(PROCESSENTRY32);
 
                 if (good)
-                    starttime = GetTickCount();
+                    starttime = (DWORD)GetTickCount64();
                 if (process_first(handle, &p))
                     do
                         RAND_add(&p, p.dwSize, 9);
                     while (process_next(handle, &p)
                            && (!good
-                               || (GetTickCount() - starttime) < MAXDELAY));
+                               || ((DWORD)GetTickCount64() - starttime) < MAXDELAY));
 
                 /* thread walking */
                 /*
@@ -535,13 +535,13 @@ int RAND_poll(void)
                  */
                 t.dwSize = sizeof(THREADENTRY32);
                 if (good)
-                    starttime = GetTickCount();
+                    starttime = (DWORD)GetTickCount64();
                 if (thread_first(handle, &t))
                     do
                         RAND_add(&t, t.dwSize, 6);
                     while (thread_next(handle, &t)
                            && (!good
-                               || (GetTickCount() - starttime) < MAXDELAY));
+                               || ((DWORD)GetTickCount64() - starttime) < MAXDELAY));
 
                 /* module walking */
                 /*
@@ -550,13 +550,13 @@ int RAND_poll(void)
                  */
                 m.dwSize = sizeof(MODULEENTRY32);
                 if (good)
-                    starttime = GetTickCount();
+                    starttime = (DWORD)GetTickCount64();
                 if (module_first(handle, &m))
                     do
                         RAND_add(&m, m.dwSize, 9);
                     while (module_next(handle, &m)
                            && (!good
-                               || (GetTickCount() - starttime) < MAXDELAY));
+                               || ((DWORD)GetTickCount64() - starttime) < MAXDELAY));
                 if (close_snap)
                     close_snap(handle);
                 else
@@ -665,7 +665,7 @@ static void readtimer(void)
     }
 
     if (!have_tsc && !have_perfc) {
-        w = GetTickCount();
+        w = (DWORD)GetTickCount64();
         RAND_add(&w, sizeof(w), 0);
     }
 }
diff --git a/ms/winrtdef.h b/ms/winrtdef.h
index b1a7598..d23ae37 100644
--- a/ms/winrtdef.h
+++ b/ms/winrtdef.h
@@ -16,10 +16,6 @@
 #  undef FindFirstFile
 # endif
 # define FindFirstFile(lpFileName, lpFindFileData) FindFirstFileEx(lpFileName, FindExInfoStandard, lpFindFileData, FindExSearchNameMatch, NULL, 0);
-# ifdef GetTickCount
-#  undef GetTickCount
-# endif
-# define GetTickCount winrt_GetTickCount
 # ifdef LoadLibraryA
 #  undef LoadLibraryA
 # define LoadLibraryA winrt_LoadLibraryA
