diff --git a/libcpluff/internal.h b/libcpluff/internal.h
index 6b16468..4a92285 100644
--- a/libcpluff/internal.h
+++ b/libcpluff/internal.h
@@ -84,7 +84,17 @@ extern "C" {
  * Macros
  * ----------------------------------------------------------------------*/
 
-#if defined(DLOPEN_POSIX)
+#if defined(_WIN32)
+#define DLHANDLE void *
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
+#define DLOPEN(name) LoadPackagedLibrary(name, 0)
+#else
+#define DLOPEN(name) LoadLibraryA(name)
+#endif
+#define DLSYM(handle, symbol) GetProcAddress(handle, symbol)
+#define DLCLOSE(handle) CloseHandle(handle)
+#define DLERROR() "WIN32 - TODO"
+#elif defined(DLOPEN_POSIX)
 #define DLHANDLE void *
 #define DLOPEN(name) dlopen((name), RTLD_LAZY | RTLD_GLOBAL)
 #define DLSYM(handle, symbol) dlsym((handle), (symbol))
diff --git a/libcpluff/pcontrol.c b/libcpluff/pcontrol.c
index c1154c7..9e1ea49 100644
--- a/libcpluff/pcontrol.c
+++ b/libcpluff/pcontrol.c
@@ -40,7 +40,9 @@
 #include "defines.h"
 #include "util.h"
 #include "internal.h"
-
+#ifdef _WIN32
+#include <windows.h>
+#endif
 
 /* ------------------------------------------------------------------------
  * Function definitions
@@ -330,7 +332,13 @@ static int resolve_plugin_runtime(cp_context_t *context, cp_plugin_t *plugin) {
 		strcpy(rlpath + ppath_len + 1 + lname_len, CP_SHREXT);
 		
 		// Open the plug-in runtime library 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
+		WCHAR wPath[MAX_PATH];
+		MultiByteToWideChar(CP_ACP, 0, rlpath, -1, (LPWSTR)wPath, MAX_PATH);
+		plugin->runtime_lib = DLOPEN(wPath);
+#else
 		plugin->runtime_lib = DLOPEN(rlpath);
+#endif
 		if (plugin->runtime_lib == NULL) {
 			const char *error = DLERROR();
 			if (error == NULL) {
diff --git a/libcpluff/psymbol.c b/libcpluff/psymbol.c
index 93c9f12..0dfe408 100644
--- a/libcpluff/psymbol.c
+++ b/libcpluff/psymbol.c
@@ -37,7 +37,9 @@
 #include "defines.h"
 #include "internal.h"
 #include "util.h"
-
+#ifdef _WIN32
+#include <windows.h>
+#endif
 
 /* ------------------------------------------------------------------------
  * Data structures
